rust/sdk-processor/src/parquet_processors/mod.rs

```diff
@@ -56,84 +86,136 @@ pub enum ParquetTypeEnum {
     MoveModule,
 }
 
-#[derive(Clone, Debug, strum::EnumDiscriminants)]
-#[strum(serialize_all = "snake_case")]
-#[strum_discriminants(
-    derive(
-        Deserialize,
-        Serialize,
-        strum::EnumVariantNames,
-        strum::IntoStaticStr,
-        strum::Display,
-        clap::ValueEnum
-    ),
-    name(ParquetTypeStructName),
-    clap(rename_all = "snake_case"),
-    serde(rename_all = "snake_case"),
-    strum(serialize_all = "snake_case")
-)]
-pub enum ParquetTypeStructs {
-    MoveResource(Vec<MoveResource>),
-    WriteSetChange(Vec<WriteSetChangeModel>),
-    Transaction(Vec<ParquetTransaction>),
-    TableItem(Vec<TableItem>),
-    MoveModule(Vec<MoveModule>),
+/// Trait for handling various Parquet types.
+#[async_trait]
+pub trait ParquetTypeTrait: std::fmt::Debug + Send + Sync + Downcast {
+    fn parquet_type(&self) -> ParquetTypeEnum;
+    fn calculate_size(&self) -> usize;
+
+    fn append(&mut self, other: Box<dyn ParquetTypeTrait>) -> Result<(), ProcessorError>;
+
+    async fn upload_to_gcs(
+        &self,
+        uploader: &mut GCSUploader,
+        parquet_type: ParquetTypeEnum,
+        table_name: &str,
+    ) -> anyhow::Result<()>;
+
+    fn clone_box(&self) -> Box<dyn ParquetTypeTrait>;
+}
+
+impl_downcast!(ParquetTypeTrait);
+
+/// Struct for handling Parquet types dynamically.
+#[derive(Debug)]
+pub struct ParquetTypeStructs {
```

Comment:
Is this struct `ParquetTypeStructs` needed if it only contains 1 field: `data`? Maybe we can use `Box<dyn ParquetTypeTrait>` directly.